### PR TITLE
Backport(5.0): Better error handling with auth connectors (#5599)

### PR DESCRIPTION
**Pull request by kimlisa**
#### Description
backport of #5599 into branch/5.0

### CODE REVIEW

**Comment by russjones** [comment]
@kimlisa Please don't merge this until March 1st.

File: lib/web/apiserver.go [hunk 1]
```diff
@@ -827,78 +827,56 @@ func (h *Handler) jwks(w http.ResponseWriter, r *http.Request, p httprouter.Para
 	return &resp, nil
 }
 
-func (h *Handler) oidcLoginWeb(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
+func (h *Handler) oidcLoginWeb(w http.ResponseWriter, r *http.Request, p httprouter.Params) string {
```
Comment (a-palchikov) [inline lib/web/apiserver.go:830]: nit: can the return value be named to it's clear it's a redirect? Something like:
```go
func (h *Handler) oidcLoginWeb(w http.ResponseWriter, r *http.Request, p httprouter.Params) (redirectURL string) {
```
should suffice

File: lib/web/apiserver.go [hunk 2]
```diff
@@ -928,39 +906,39 @@ func (h *Handler) githubLoginConsole(w http.ResponseWriter, r *http.Request, p h
...
+		return client.LoginFailedBadCallbackRedirectURL
 	}
+
 	// if we created web session, set session cookie and redirect to original url
 	if response.Req.CreateWebSession {
-		err = csrf.VerifyToken(response.Req.CSRFToken, r)
-		if err != nil {
-			logger.Warnf("Unable to verify CSRF token: %v.", err)
-			return nil, trace.AccessDenied("access denied")
+		logger.Infof("Redirecting to web browser.")
```
Comment (a-palchikov) [inline lib/web/apiserver.go:921]:
```suggestion
		logger.Info("Redirecting to web browser.")
```

File: lib/web/apiserver.go [hunk 3]
```diff
@@ -974,10 +952,11 @@ func (h *Handler) githubCallback(w http.ResponseWriter, r *http.Request, p httpr
 		FIPS:              h.cfg.FIPS,
 	})
 	if err != nil {
-		return nil, trace.Wrap(err)
+		logger.WithError(err).Error("Error constructing ssh response")
```
Comment (a-palchikov) [inline lib/web/apiserver.go:955]:
```suggestion
		logger.WithError(err).Error("Error constructing ssh response.")
```

**Review by a-palchikov — APPROVED** [review]

**Comment by kimlisa** [comment]
backporting is frozen atm

**Pull request closed without merging**